sdk/identity/Azure.Identity/src/DefaultAzureCredential.cs

```diff
@@ -138,21 +152,27 @@ private async ValueTask<AccessToken> GetTokenFromSourcesAsync(bool async, TokenR
 
                     return token;
                 }
-                catch (CredentialUnavailableException e)
+                catch (AuthenticationFailedException e)
                 {
                     exceptions.Add(e);
                 }
             }
 
-            // build the credential unavailable message, this code is only reachable if all credentials throw CredentialUnavailableException
+            // Build the credential unavailable message, this code is only reachable if all credentials throw AuthenticationFailedException
             StringBuilder errorMsg = new StringBuilder(DefaultExceptionMessage);
 
-            foreach (Exception ex in exceptions)
+            bool allCredentialUnavailableException = true;
+            foreach (AuthenticationFailedException ex in exceptions)
             {
-                errorMsg.Append(Environment.NewLine).Append(ex.Message);
+                allCredentialUnavailableException &= ex is CredentialUnavailableException;
+                errorMsg.Append(Environment.NewLine).Append("- ").Append(ex.Message);
             }
 
-            throw new CredentialUnavailableException(errorMsg.ToString());
+            // If all credentials have thrown CredentialUnavailableException, throw CredentialUnavailableException,
+            // otherwise throw AuthenticationFailedException
+            throw allCredentialUnavailableException
```

Comment:
This change fixes #13741 and other similar issues. While `AuthenticationFailedException` is valid, we would like to ignore it if another `TokenCredential` has succeeded. However, if all source credentials have failed, we should throw `AuthenticationFailedException` instead of `CredentialUnavailableException`.